engine/bsql_engine/io/cio.pxd

```diff
@@ -144,29 +144,28 @@ cdef extern from * namespace "blazing":
         template <class T> inline typename std::remove_reference<T>::type&& blaz_move(T&& t) { return std::move(t); }
         }
         """
-        cdef T blaz_move[T](T)
+        cdef T blaz_move[T](T) nogil
```

Comment:
this is a pretty trivial function and the overhead of releasing and reacquiring the gil is probably more than just allowing it to use the gil while calling move

Reply:
Yes I can remove that. FYI, a `nogil` function annotation doesn't actually release the GIL, it just doesn't attempt to acquire it: https://cython.readthedocs.io/en/latest/src/userguide/external_C_code.html?highlight=nogil#declaring-a-function-as-callable-without-the-gil
```
The nogil function annotation declares that it is safe to call the function without the GIL. It 
is perfectly allowed to execute it while holding the GIL. The function does not in itself release 
the GIL if it is held by the caller. 
```
Is that helpful or would you still prefer it be removed?

Reply:
Turns out `blaz_move()` has to be declared with `nogil` if it's to be called from within a `with nogil ...` block.

Reply:
That's fine leave it with nogil